uxarray/grid.py

```diff
@@ -0,0 +1,279 @@
+
+        # determine initialization type - string signifies a file, numpy array signifies a list of verts
+        in_type = type(args[0])
+
+        # check if initializing from verts:
+        try:
+            if os.path.isfile(args[0]) is False and in_type is not np.ndarray:
+                raise FileNotFoundError("File not found: " + args[0])
+            elif in_type is np.ndarray:
+                self.vertices = args[0]
+                self.from_vert()
+        except ValueError as e:
+            # initialize from vertices
+            if in_type is np.ndarray:
+                self.vertices = args[0]
+                self.from_vert()
+            else:
+                print(
+                    "Init called with args other than verts (ndarray) or filename (str)"
+                )
+                exit()
+
+        # check if initialize from file:
+        if in_type is str and os.path.isfile(args[0]):
+            self.filepath = args[0]
+            self.from_file()
+
+        # initialize for gridspec
+        elif in_type is str:
+            self.gridspec = args[0]
+            self.from_gridspec()
+
+        else:
+            # this may just be local initialization for with no options or options other than above
+            pass
+
+    # vertices init
+    def from_vert(self):
+        """Create a grid with one face with vertices specified by the given
+        argument."""
+        self._init_mesh2()
+        self.in_ds.Mesh2.attrs['topology_dimension'] = self.vertices[0].size
+
+        self.in_ds["Mesh2"].topology_dimension
+
+        x_coord = self.vertices.transpose()[0]
+        y_coord = self.vertices.transpose()[1]
+
+        # single face with all nodes
+        num_nodes = x_coord.size
+        conn = list(range(0, num_nodes))
+        conn = [conn]
+
+        self.in_ds["Mesh2_node_x"] = xr.DataArray(data=xr.DataArray(x_coord),
+                                                  dims=["nMesh2_node"])
+        self.in_ds["Mesh2_node_y"] = xr.DataArray(data=xr.DataArray(y_coord),
+                                                  dims=["nMesh2_node"])
+        self.in_ds["Mesh2_face_nodes"] = xr.DataArray(
+            data=xr.DataArray(conn),
+            dims=["nMesh2_face", "nMaxMesh2_face_nodes"],
+            attrs={
+                "cf_role": "face_node_connectivity",
+                "_FillValue": -1,
+                "start_index": 0
+            })
+
+    # TODO: gridspec init
+    def from_gridspec(self):
+        print("initializing with gridspec")
+
+    # load mesh from a file
+    def from_file(self):
+        """Loads a mesh file Also, called by __init__ routine This routine will
+        automatically detect if it is a UGrid, SCRIP, Exodus, or shape file.
+
+        Raises:
+            RuntimeError: Invalid file type
+        """
+        # find the file type
+        self.meshFileType = self.find_type()
+
+        # call reader as per meshFileType
+        if self.meshFileType == "exo":
```

Comment:
Variable naming with mixed cases and combined words (e.g. meshFileType) is not recommended in [PEP8](https://www.python.org/dev/peps/pep-0008/#:~:text=module%20non%2Dpublic%22).-,Function%20and%20Variable%20Names,Variable%20names%20follow%20the%20same%20convention%20as%20function%20names.,-mixedCase%20is%20allowed). Furthermore, I think we need to follow [PEP8 instance variable naming conventions](https://www.python.org/dev/peps/pep-0008/#:~:text=Use%20one%20leading%20underscore%20only%20for%20non%2Dpublic%20methods%20and%20instance%20variables.) for this variable as it is an instance variable.